oc_cdtapi/DmsGetverAPI.py

```diff
@@ -46,6 +93,92 @@ def create_distr_request(self, version=None, source_version=None, distr_type=Non
+        headers = self.get_headers()
+        url = posixpath.join('api', 'v1', 'images', image_id, 'download')
+        logging.debug('url: [%s]' % url)
+        tf = tempfile.NamedTemporaryFile(delete=False)
+        tfn = tf.name
+        logging.debug('tf: [%s]' % tfn)
+        r = self.get(url, headers=headers, stream=True)
+        for chunk in r.iter_content(chunk_size=8192):
+            tf.write(chunk)
+        tf.seek(0)
+        return tfn
+
+    def get_audit(self, audit_id):
+        """
+        Requests audit record by record id
+        :param str audit_id: id of audit record
+        :return: audit record
+        """
+        logging.debug('Reached get_audit')
+        logging.debug('audit_id: [%s]' % audit_id)
+        self.web.auth = None
+        self.raise_exception_high = 499
+        headers = self.get_headers()
+        url = posixpath.join('api', 'v1', 'audit', audit_id)
+        r = self.get(url, headers=headers)
+        return self.json_or_none(r)
+
+    def get_headers(self):
+        """
+        """
+        logging.debug('Reached get_headers')
+        logging.debug('auth_token: [%s]' % self.auth_token)
+        headers = {'Accept': 'application/json; charset=utf-8', 'Authorization': f'Bearer {self.auth_token}'}
+        logging.debug('About to return: [%s]' % headers)
+        return headers
+
+    def search_image(self, version=None, distr_type=None):
+        """
+        searches for images, calls download
+        :return: path to temp file, image data
+        """
+        logging.debug('Reached download_image')
+        logging.debug('Unsetting auth')
+        self.web.auth = None
+        logging.debug('version = [%s]' % version)
+        logging.debug('distr_type = [%s]' % distr_type)
+        
+        url = posixpath.join('api', 'v1', 'images')
+        headers = self.get_headers()
+        params = {
+            'strict_filters': 'true',
+            'product_type': distr_type.lower(),
+            'product_version': version,
+            'product_release_stage': 'release',
+            'remap_tablespaces': 'true'
+        }
+        resp = self.get(url, headers=headers, params=params)
+        if not resp.status_code == 200:
+            logging.error('Server returned an error [%s] [%s]' % (resp.status_code, resp.text))
+            return None
+        images = resp.json()['items']
+        logging.debug('found [%s] images' % len(images))
+        logging.debug('Dumping images data')
+        logging.debug(json.dumps(images, indent=4))
+        for image in images:
+            image_id = image['id']
+            image_name = image['name']
+            oracle_version = image.get('oracle_version')
+            customisation = image.get('customisation')
+            logging.debug('checking images [%s]' % image_name)
+            if oracle_version['oracle_edition'] == 'EE' and customisation is None:
+                logging.debug('image is enterprise edition')
+                return image
```

Comment:
Too  much hardcoded values.
1. You have specified `self.oracle_edition` in the constructor but using hardcoded 'EE' here instead of that.
2. We need a possibility to filter images by other oracle-related parameters (like `oracle_version`, `codepage` and so on), if it is possible. `oracle_edition` have to be re-definable also.

Reply:
fixed